Update to Google Trusty April code base

AjanZhong opened the pull request:
Following features are unsupported on this release, they are still working in progress:
    keymint TA support.
    Rust support.
    Binder support.

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>

Commit:
Update to Google Trusty April code base

DO NOT MERGE, WIP
Following features are unsupported on this release, they are still
working in progress:
    keymint TA support.
    Rust support.
    Binder support.

Signed-off-by: Zhong,Fangjian <fangjian.zhong@intel.com>

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>

sysopenci commented:
Expected syntax:
    Commit message one-line summary

      Extra paragraphs can be added here.

      Longer description that explains the problem / solution
      or whatever that needs to be said

      It can contain
        multiple lines
        multiple paragraphs
        blank lines
        references

      But each paragraph must be followed by a blank line

      However, you need to keep the 'Tracked-On', 'Change-Id',
      'Signed-off-by', ..., fields in the **lastest** paragraph of
      the commit message.

      Change-Id: I9e58f79c7faf7f0edb9792e8b5322ab161331141
      Tracked-On: https://jira.server.ext/browse/PROJECTA-123
      Signed-off-by: Developer Name <developer.name@intel.com>
